apps/api/test/api/client/channel_test.exs

```diff
@@ -1,26 +1,228 @@
+      ref = push(socket, "list_relays", %{"resource_id" => resource.id})
+      assert_reply ref, :error, :offline
+    end
+
+    test "returns list of online relays", %{account: account, resource: resource, socket: socket} do
+      relay = RelaysFixtures.create_relay(account: account)
+      stamp_secret = Ecto.UUID.generate()
+      :ok = Domain.Relays.connect_relay(relay, stamp_secret)
+
+      ref = push(socket, "list_relays", %{"resource_id" => resource.id})
+      assert_reply ref, :ok, %{relays: relays}
+
+      ipv4_uri = "stun:#{relay.ipv4}:#{relay.port}"
+      ipv6_uri = "stun:#{relay.ipv6}:#{relay.port}"
+
+      assert [
+               %{
+                 expires_at: expires_at,
+                 password: password1,
+                 username: username1,
+                 uri: ^ipv4_uri
+               },
+               %{
+                 expires_at: expires_at,
+                 password: password2,
+                 username: username2,
+                 uri: ^ipv6_uri
+               }
+             ] = relays
+
+      assert username1 != username2
+      assert password1 != password2
+
+      assert [expires_at, salt] = String.split(username1, ":", parts: 2)
+      expires_at = expires_at |> String.to_integer() |> DateTime.from_unix!()
+      socket_expires_at = DateTime.truncate(socket.assigns.expires_at, :second)
+      assert expires_at == socket_expires_at
+
+      assert is_binary(salt)
+    end
+  end
+
+  describe "handle_in/3 request_connection" do
+    test "returns error when resource is not found", %{socket: socket} do
+      attrs = %{
+        "resource_id" => Ecto.UUID.generate(),
+        "client_rtc_session_description" => "RTC_SD",
+        "client_preshared_key" => "PSK"
+      }
+
+      ref = push(socket, "request_connection", attrs)
+      assert_reply ref, :error, :not_found
+    end
+
+    test "returns error when all gateways are offline", %{
+      resource: resource,
+      socket: socket
+    } do
+      attrs = %{
+        "resource_id" => resource.id,
+        "client_rtc_session_description" => "RTC_SD",
+        "client_preshared_key" => "PSK"
+      }
+
+      ref = push(socket, "request_connection", attrs)
+      assert_reply ref, :error, :offline
+    end
+
+    test "returns error when all gateways connected to the resource are offline", %{
+      account: account,
+      resource: resource,
+      socket: socket
+    } do
+      attrs = %{
+        "resource_id" => resource.id,
+        "client_rtc_session_description" => "RTC_SD",
+        "client_preshared_key" => "PSK"
+      }
+
+      gateway = GatewaysFixtures.create_gateway(account: account)
+      :ok = Domain.Gateways.connect_gateway(gateway)
+
+      ref = push(socket, "request_connection", attrs)
+      assert_reply ref, :error, :offline
+    end
+
+    test "broadcasts request_connection to the gateways and then returns connect message", %{
```

Comment:
@conectado This is the main test that showcases how the connection is happening for the client.